src/homeworkNumber5/bracketBalance/advancedBracketBalance.c

```diff
@@ -0,0 +1,69 @@
+#include "stack.h"
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+bool isBalanced(const char* str)
+{
+    Stack* stack = createStack();
+    if (stack == NULL)
+        return false;
+
+    for (int i = 0; str[i] != '\0'; i++) {
+        char current = str[i];
+
+        bool isOpen = (current == '(' || current == '[' || current == '{');
+
+        if (isOpen) {
+            push(stack, current);
+        } else if (current == ')' || current == ']' || current == '}') {
+            if (isEmpty(stack)) {
+                deleteStack(stack);
+                free(stack);
+                return false;
+            }
+
+            int top;
+            if (pop(stack, &top) != 0) {
+                deleteStack(stack);
+                free(stack);
+                return false;
+            }
+
+            // Проверяем соответствие скобок
+            if ((current == ')' && top != '(') || (current == ']' && top != '[') || (current == '}' && top != '{')) {
+                deleteStack(stack);
+                free(stack);
+                return false;
+            }
+        }
+    }
+
+    // Если после обработки всей строки стек пуст - сбалансировано
+    bool result = isEmpty(stack);
+    deleteStack(stack);
```

Comment:
```suggestion
    deleteStack(stack);
    free(stack);
```
А ещё можно сделать флажок, чтобы этот код не дублировать